app/src/main/java/org/eyeseetea/malariacare/presenter/ReviewPresenter.java

```diff
@@ -0,0 +1,122 @@
+package org.eyeseetea.malariacare.presenter;
+
+import com.google.common.collect.Iterables;
+
+import org.eyeseetea.malariacare.DashboardActivity;
+import org.eyeseetea.malariacare.data.database.model.QuestionRelation;
+import org.eyeseetea.malariacare.data.database.model.Survey;
+import org.eyeseetea.malariacare.data.database.utils.Session;
+import org.eyeseetea.malariacare.domain.entity.Value;
+import org.eyeseetea.malariacare.layout.adapters.dashboard.ReviewScreenAdapter;
+import org.eyeseetea.malariacare.strategies.ReviewFragmentStrategy;
+import org.eyeseetea.malariacare.utils.Constants;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class ReviewPresenter implements ReviewScreenAdapter.onClickListener {
+
+    public interface ReviewView {
+        void showValues(List<Value> values);
+
+        void initListView();
+    }
+
+    ReviewView view;
+
+    public ReviewPresenter() {
+    }
+
+    public void attachView(ReviewView reviewView) {
+        this.view = reviewView;
+        view.showValues(prepareValues());
+        view.initListView();
+    }
+
+    private List<org.eyeseetea.malariacare.domain.entity.Value> prepareValues() {
+        Iterator<String> colorIterator;
+        List<org.eyeseetea.malariacare.domain.entity.Value> preparedValues = new ArrayList<>();
+        List<org.eyeseetea.malariacare.data.database.model.Value> values;
+        values = getReviewValues();
```

Comment:
Presenter only should know domain entities and not data models, can you create a GetReviewValuesBySurveyIdUseCase that returns a list of domain Value and inject this new use case in constructor?